tests/test_threadpoolctl.py

```diff
@@ -302,6 +302,9 @@ def test_nested_prange_blas(nthreads_outer):
     assert original_info == _threadpool_info()
 
 
+# the method `get_original_num_threads` raises a UserWarning in one CI job. It
+# is expected so we can safely filter it.
+@pytest.mark.filterwarnings("ignore::UserWarning")
```

Comment:
Is there a way to filter a warning based on the expected warning message ?
I tried something like "ignore:Found" because 'Found' is in the warning message but it does not work.

Reply:
You can use the context manager: 

https://docs.pytest.org/en/3.0.5/recwarn.html#recwarn

and then make assertions, ignore or raise based on the warning message contents.

Reply:
yes but then pytest fails if there's no warning inside the context manager.

Reply:
The `recwarn` fixture I linked above does not fail if so warning is raised.

Reply:
But this is fine as it is anyway, merging.